### PR TITLE
Add New Relic support

### LINKED ISSUE #3: New Relic
Easy interface for integrating New Relic APM: https://docs.newrelic.com/docs/agents/nodejs-agent/installation-configuration/install-nodejs-agent

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #3 

This basically adds some sanity checks before requiring the `newrelic` library. We'll need the following as env variables for each container:

```
NEW_RELIC_APP_NAME  // (optional) The name that shows in New Relic (default in newrelic is My Application)
NEW_RELIC_LICENSE_KEY // (required) License key to use
NEW_RELIC_NO_CONFIG_FILE // (required) Should be true to use env variables instead of config file
```

Is there somewhere where we need to properly set those? I can take care of the implementation. (or we just write it down for now?)

This PR still needs:
- [ ] Implement (or note) env variables to set
- [x] README for New Relic
- [x] Expose it in `index`

### CODE REVIEW

**Comment by mjangda** [comment]
> Is there somewhere where we need to properly set those?

This can be done via `vipd` (see `libs/class-nodejs-app-config.php`).

File: src/newrelic/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+	const licenseKey = process.env.NEW_RELIC_LICENSE_KEY;
+	const noConfig = process.env.NEW_RELIC_NO_CONFIG_FILE;
+	const isProduction = () => process.env.NODE_ENV === 'production';
+
+	if ( isProduction && noConfig && licenseKey ) {
+		logger.info( 'Importing New Relic library...' );
+		return require( 'newrelic' );
+	}
+
+	logger.error( 'New Relic could not be loaded, environment variables are not present' );
```
Comment (mjangda) [inline src/newrelic/index.js]: Would be helpful to note which environment variables are missing.

File: src/newrelic/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = ( { logger = console } = {} ) => {
+	const licenseKey = process.env.NEW_RELIC_LICENSE_KEY;
+	const noConfig = process.env.NEW_RELIC_NO_CONFIG_FILE;
+	const isProduction = () => process.env.NODE_ENV === 'production';
```
Comment (mjangda) [inline src/newrelic/index.js]: I don't think we need the `NODE_ENV` check. We may have a non-production environment hosted on Go as well.

File: src/newrelic/index.js (hunk 1 above)
Comment (mjangda) [inline src/newrelic/index.js]: If we add an error code like `#missing-env-vars` to the end of the error we can check for that in our test as well to make sure we're triggering the right error.

File: __tests__/newrelic.tests.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+test( 'New Relic module should fail if no license key is given', () => {
+	// eslint-disable-next-line no-undef
+	const spyOnError = jest.spyOn( console, 'error' );
+
+	newrelic();
+
+	expect( spyOnError ).toHaveBeenCalled();
+
+	spyOnError.mockRestore();
+} );
```
Comment (mjangda) [inline __tests__/newrelic.tests.js]: For the "success" test case, we cna probably use jest's auto-mock support: https://jestjs.io/docs/en/manual-mocks#mocking-node-modules

Basically, add a path called `__mocks__/newrelic.js` and we have it export a value we can verify in our test.

Comment (elazzabi) [reply __tests__/newrelic.tests.js]: Never heard of the auto mocking, that's actually very helpful. Thanks 🙌

**Comment by elazzabi** [comment]
Fixed all the notes, still needs vars implementation on `vipd` and a `README`

File: __tests__/newrelic.tests.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+const newrelic = require( '../src/newrelic/' );
+
+describe( 'Should fail if some environment variables are not present', () => {
```
Comment (mjangda) [inline __tests__/newrelic.tests.js]: Minor: Ideally, we should wrap tests in a describe with the module name, i.e.

```
describe( 'src/newrelic', () => {
    describe( 'Should fail if some environment variables are not present', () => {
    ...
} );
```

For modules, with lots of exported values/methods/etc., it's also nice to break down further by the method-under-test:

```
describe( 'src/newrelic', () => {
    describe( 'newrelic()', () => {
        describe( 'should fail if env vars missing', () => {
            test( ... );
        } ); 
    } );
} );
```

Comment (elazzabi) [reply __tests__/newrelic.tests.js]: Oh, right, that would be nice. I'll work on it (and probably updating the others too)

**Pull request merged**